libsplinter/src/service/scabbard/client/error.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::error::Error as ErrorTrait;
+use std::fmt;
+
+#[derive(Debug)]
+pub struct Error(pub String);
```

Comment:
It's not clear to me yet if it would be useful to enumerate this error.

Reply:
I think the source pattern we've used in a few places could be useful.

```
struct Error {
   source: Box<dyn Error>,
   message: String
}
```